php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1498 @@
+    return b.<ComputedVariableTree>nonterminal(Kind.COMPUTED_VARIABLE_NAME)
+      .is(f.computedVariableName(
+        b.token(LCURLYBRACE),
+        EXPRESSION(),
+        b.token(RCURLYBRACE)));
+
+  }
+
+  public VariableIdentifierTree VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(Kind.VARIABLE_IDENTIFIER)
+      .is(f.variableIdentifier(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  public VariableTree COMPOUND_VARIABLE() {
+    return b.<VariableTree>nonterminal(Kind.COMPOUND_VARIABLE_NAME)
+      .is(
+        b.firstOf(
+          VARIABLE_IDENTIFIER(),
+          f.compoundVariable(b.token(DOLLAR_LCURLY), EXPRESSION(), b.token(RCURLYBRACE))));
+  }
+
+  public ExpressionTree VARIABLE_WITHOUT_OBJECTS() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.VARIABLE_WITHOUT_OBJECTS)
+      .is(f.variableWithoutObjects(
+        b.zeroOrMore(b.token(PHPLexicalGrammar.VARIABLE_VARIABLE_DOLLAR)),
+        COMPOUND_VARIABLE(),
+        b.zeroOrMore(b.firstOf(
+          DIMENSIONAL_OFFSET(),
+          ALTERNATIVE_DIMENSIONAL_OFFSET()
+        ))));
+  }
+
+  public ArrayAccessTree ALTERNATIVE_DIMENSIONAL_OFFSET() {
+    return b.<ArrayAccessTree>nonterminal()
+      .is(f.alternativeDimensionalOffset(
+        b.token(LCURLYBRACE),
+        b.optional(EXPRESSION()),
+        b.token(RCURLYBRACE)
+      ));
+  }
+
+  public ArrayAccessTree DIMENSIONAL_OFFSET() {
+    return b.<ArrayAccessTree>nonterminal(PHPLexicalGrammar.DIMENSIONAL_OFFSET)
+      .is(f.dimensionalOffset(
+        b.token(LBRACKET),
+        b.optional(EXPRESSION()),
+        b.token(RBRACKET)
+      ));
+  }
+
+  public ExpressionTree PRIMARY_EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.PRIMARY_EXPRESSION)
+      .is(
+        b.firstOf(
+          f.newStaticIdentifier(b.token(STATIC)),
+          // FIXME (Linda): add b.sequence(PHPKeyword.NAMESPACE, FULLY_QUALIFIED_CLASS_NAME)
+          VARIABLE_WITHOUT_OBJECTS(),
+          IDENTIFIER(),
+          PARENTHESIZED_EXPRESSION()
+        ));
+  }
+
+  public ReferenceVariableTree REFERENCE_VARIABLE() {
+    return b.<ReferenceVariableTree>nonterminal(Kind.REFERENCE_VARIABLE)
+      .is(f.referenceVariable(
+        b.token(AMPERSAND),
+        MEMBER_EXPRESSION()));
+  }
+
+  public SpreadArgumentTree SPREAD_ARGUMENT() {
+    return b.<SpreadArgumentTree>nonterminal(Kind.SPREAD_ARGUMENT)
+      .is(f.spreadArgument(
+        b.token(ELIPSIS),
+        EXPRESSION()
+      ));
+  }
+
+  public FunctionCallTree FUNCTION_CALL_ARGUMENT_LIST() {
+    return b.<FunctionCallTree>nonterminal(PHPLexicalGrammar.FUNCTION_CALL_PARAMETER_LIST)
+      .is(f.functionCallParameterList(
+        b.token(LPARENTHESIS),
+        b.optional(f.newTuple9(
+          FUNCTION_CALL_ARGUMENT(), b.zeroOrMore(f.newTuple5(b.token(COMMA), FUNCTION_CALL_ARGUMENT())))),
+        b.token(RPARENTHESIS)));
+  }
+
+  public ExpressionTree FUNCTION_CALL_ARGUMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)